front/lib/workspace_usage.ts

```diff
@@ -94,3 +97,28 @@ export async function unsafeGetUsageData(
 
   return csvHeader + csvContent;
 }
+
+/**
+ * Check if a workspace is active during a trial based on the following conditions:
+ *   - Existence of a connected data source
+ *   - Existence of a custom assistant
+ *   - A conversation occurred within the past 7 days
+ */
+export async function checkWorkspaceActivity(workspace: Workspace) {
+  const sevenDaysAgo = new Date();
+  sevenDaysAgo.setDate(sevenDaysAgo.getDate() - 7);
+
+  const hasConnectedDataSource = await DataSource.findOne({
```

Comment:
We want to consider as active if there's a Connection or if there's a Connection or a Folder?

Reply:
Connection and folder do create a data source 🙃.

Reply:
Seen IRL, fixed.